httpmock.go

```diff
@@ -84,12 +103,19 @@ func NewServer(handler Handler) *Server {
 	return s
 }
 
-// NewServer constructs a new server but doesn't start it (compare to httptest.NewUnstartedServer).
+// NewUnstartedServer constructs a new server but doesn't start it (compare to httptest.NewUnstartedServer).
```

Comment:
Actually could you add one quick comment in this method doc to indicate that if we're passed a handler that supports HandlerWithHeaders then it'll use that, otherwise it'll use the normal one? I realized someone could get pretty tripped up if they try to use a mock or fake that supports both calls.

Reply:
good call!